hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -514,4 +558,83 @@ bool OsmApiWriter::_uploadChangeset(HootNetworkRequestPtr request, long id, cons
   return success;
 }
 
+bool OsmApiWriter::_resolveIssues(HootNetworkRequestPtr request, ChangesetInfoPtr changeset)
+{
+  bool success = false;
+  //  Can only work on changesets of size 1
+  if (changeset->size() != 1)
+    return false;
+  for(int elementType = 0; elementType < ElementType::Type::Unknown; ++elementType)
+  {
+    for (int changesetType = 0; changesetType < XmlChangeset::ChangesetType::TypeMax; ++changesetType)
+    {
+      ChangesetInfo::iterator element = changeset->begin((ElementType::Type)elementType, (XmlChangeset::ChangesetType)changesetType);
+      if (element != changeset->end((ElementType::Type)elementType, (XmlChangeset::ChangesetType)changesetType))
+      {
+        long id = *element;
+        QString update = "";
+        //  Get the element from the OSM API
+        if (elementType == ElementType::Node)
+          update = _getNode(request, id);
+        else if (elementType == ElementType::Way)
+          update = _getWay(request, id);
+        else if (elementType == ElementType::Relation)
+          update = _getRelation(request, id);
+        //  Fix the changeset with the node/way/relation from the OSM API
+        success |= _changeset.fixChangeset(update);
+      }
+    }
+  }
+  return success;
+}
```

Comment:
Iterate all nine data structures looking for the failed element.  (add, modify, delete * node, way, relation) then query the OSM API through the `_get*` functions and try to fix the changeset.